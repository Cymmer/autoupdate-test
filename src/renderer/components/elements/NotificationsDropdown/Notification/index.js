import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

const Notification = ({ children, className, viewAction }) => (
  <button
    className={cn(styles.Notification, className)}
    data-test="notificationButton"
    type="button"
    onClick={viewAction}
  >
    {children}
  </button>
);

Notification.defaultProps = {
  className: null,
};

Notification.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,

  // opens the modal to view the notification
  viewAction: PropTypes.func.isRequired,
};

export default Notification;
