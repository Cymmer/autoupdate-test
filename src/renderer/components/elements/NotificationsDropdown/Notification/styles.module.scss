@import '../../../../scss/variables';
@import '../../../../scss/mixins';
@import '../../../../scss/themes';

.Notification {
  @include ease(0.2s);
  display: grid;
  grid-template-columns: 36px auto;
  grid-gap: 12px;
  align-items: center;
  padding: 12px 20px;
  text-align: left;
  border: none;
  border-bottom: 1px solid;
  cursor: pointer;

  @include theme {
    border-color: t('borderColor');
    background-color: t('cardBackgroundColor');

    &:hover,
    &:focus {
      background-color: t('cardBackgroundColor--hover');
    }

    &:active {
      background-color: t('cardBackgroundColor--active');
    }
  }
}
