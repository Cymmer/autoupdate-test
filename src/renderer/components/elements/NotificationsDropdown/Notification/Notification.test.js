import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import Notification from './index';

describe('RENDERING - NotificationsDropdown > Notification', () => {
  it('should render with default props', () => {
    const component = shallow(<Notification />);
    expect(component).toMatchSnapshot();
  });

  it('should be clickable', () => {
    const clickFn = jest.fn();
    const component = shallow(<Notification viewAction={clickFn} />);

    findByTestAttr(component, 'notificationButton').simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });
});
