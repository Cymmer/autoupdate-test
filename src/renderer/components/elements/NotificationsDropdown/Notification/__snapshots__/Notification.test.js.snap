// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - NotificationsDropdown > Notification should render with default props 1`] = `
<button
  className="Notification"
  data-test="notificationButton"
  type="button"
/>
`;
